sdk/python/kfp/_client.py

```diff
@@ -277,9 +289,32 @@ def set_user_namespace(self, namespace):
       namespace: kubernetes namespace the user has access to.
     """
     self._context_setting['namespace'] = namespace
+    if not os.path.exists(os.path.dirname(Client.LOCAL_KFP_CONTEXT)):
+        os.makedirs(os.path.dirname(Client.LOCAL_KFP_CONTEXT))
     with open(Client.LOCAL_KFP_CONTEXT, 'w') as f:
       json.dump(self._context_setting, f)
 
+  def get_kfp_healthz(self):
+    """Get healthz info of KFP deployment.
+
+    Returns:
+      response: json formatted response from the healtz endpoint.
+    """
+    healthz_api = 'http://' + self._existing_config.host + '/' + Client.HEALTH_PATH
```

Comment:
I believe the host should contain schema in it, some hosts may be used with `https` and some may be used with `http`, so I think we should remove `http://` here

Reply:
in fact, I think your test failure is exactly because of this: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/4638/kubeflow-pipeline-sample-test/1325844927585718272

We had a recent change to use https host for presubmit tests, so it breaks when you add `http://` there

Reply:
I was also looking at that, thinking at first `_get_url_prefix` might be an option but that sets the uihost. However, if the empty response was due to an `http` request being sent to an `https` gateway, I would expect an http error that `r.raise_for_status()` would then pick up on.

Reply:
I must have written my response just before your second message so I hadn't seen it yet. If you meant that `self._existing_config.host` should already contain `http://` or `https://`, on my cluster it doesn't but that might be outdated as it is kubeflow 1.1. I just pushed a commit changing to `https` to test that theory and hopefully clear things up. If that comes back positive I'll need to figure out a method for determining if it should use `http` or `https`.

Reply:
@Bobgy  I think I may have found the problem. In the log it says `KFP API host is 502b4a4db4ab8efb-dot-datalab-vm-staging.googleusercontent.com`, however, due to forcing https the request got an SSL error when it tried to connect to `ml-pipeline.kubeflow.svc.cluster.local`. Is there a way to get the API host other than using `self._existing_config.host`, as it seems the in_cluster host is not returning a response?

Reply:
While trying to figure out where I would be able to grab `502b4a4db4ab8efb-dot-datalab-vm-staging.googleusercontent.com` from I noticed that in the `run_sample_test.py` the host is set to `'ml-pipeline.%s.svc.cluster.local:8888' % self._namespace`. Is that expected behavior given the above statement that presumbit tests have been changed to `https`?

Reply:
As far as I can tell, requests is handling the redirection from `http` to `https` properly. The final url it lands on is `https://accounts.google.com/ServiceLogin?continue=https%3A%2F%2Fdatalab-staging.cloud.google.com%2F_signin%3Fcontinue%3Dhttps%25253A%25252F%25252F72a2c93edd5b49f8-dot-datalab-vm-staging.googleusercontent.com%25252Fapis%25252Fv1beta1%25252Fhealthz%26endpoint%3D72a2c93edd5b49f8`, which means the above statement that it is using `ml-pipeline.kubeflow.svc.cluster.local` for the API address is not correct. The header that is returned is:
```
{'Content-Type': 'text/html; charset=UTF-8', 'X-Frame-Options': 'DENY', 'x-auto-login': 'realm=com.google&args=continue%3Dhttps%253A%252F%252Fdatalab-staging.cloud.google.com%252F_signin%253Fcontinue%253Dhttps%2525253A%2525252F%2525252F72a2c93edd5b49f8-dot-datalab-vm-staging.googleusercontent.com%2525252Fapis%2525252Fv1beta1%2525252Fhealthz%2526endpoint%253D72a2c93edd5b49f8', 'Cache-Control': 'no-cache, no-store, max-age=0, must-revalidate', 'Pragma': 'no-cache', 'Expires': 'Mon, 01 Jan 1990 00:00:00 GMT', 'Date': 'Wed, 11 Nov 2020 14:49:06 GMT', 'Transfer-Encoding': 'chunked', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'Content-Security-Policy': "script-src 'nonce-RHIL2rIpv6K3J5D6qGt4lQ' 'unsafe-inline' 'unsafe-eval';object-src 'none';base-uri 'self';report-uri /cspreport", 'X-Content-Type-Options': 'nosniff', 'X-XSS-Protection': '1; mode=block', 'Server': 'GSE', 'Set-Cookie': '__Host-GAPS=1:2RJhKy8dLzlqLoYcVdWNgDaydS-zkA:HWtOcjrS-7olt-5P;Path=/;Expires=Fri, 11-Nov-2022 14:49:06 GMT;Secure;HttpOnly;Priority=HIGH'}
```

The response content seems to be a google login page or something similar. 

@Bobgy Do you know the type of restrictions there are on accessing the API endpoint? Does a specific header or cookie need to be passed along for example?